js/src/logger.ts

```diff
@@ -675,6 +676,114 @@ export interface LogOptions<IsAsyncFlush> {
 
 export type PromiseUnless<B, R> = B extends true ? R : Promise<Awaited<R>>;
 
+/**
+ * A wrapper object that allows lazily uploading attachments and their metadata. This object can be logged in place of an attachment that is being uploaded asynchronously.
+ */
+export class LazyAttachmentUpload {
```

Comment:
I think for the user facing class, we should just call it `Attachment`. All of our classes are lazy, so no need to specify that explicitly.